cpp/ql/test/query-tests/Security/CWE/CWE-119/semmle/tests/tests.cpp

```diff
@@ -654,6 +654,37 @@ void test26(bool cond)
 	if (ptr[-1] == 0) { return; } // GOOD: accesses buffer[1]
 }
 
+void test27(){
+	char *src = "";
+	char *dest = "abcdefgh";
+	const int size = 100;
+	int ind = 100;
+	char buffer[size];
+
+	strncpy(dest, src, strlen(dest)); // GOOD, strncpy will not read past null terminator of source
```

Comment:
I don't get a result here with or without the change to `StrncpyBA`.

Reply:
Hmm. That's odd. Let me re-assess this.

Reply:
This is my fault. I added the test case and observed it corrected the real-world examples, and it did cause a deviation in the expected files, but not in what was ultimately selected/alerted. 

Digging on this a bit more, I'm currently unable to get this to reproduce in a unit test, (and the other one mentioned above regarding the reachable change) but I'm seeing FPs all over the place in production in real-world code. I'll have to hit the drawing board. Maybe I can figure something out. 

In the meantime, we are moving forward with these changes, as an FYI, as it is severely biting real-world repositories. I wonder if there isn't some deeper issue. 

You'll note one of my examples does repro in the unit test (the one demonstrating the type check issue). I added that one yesterday before making this PR and the only way I could get it to reproduce was by having more complex structure in the MYSTRUCT (the fields specifically). Without that, that test would also fail to repro, and I don't know why. I would have thought the fields have no impact on that change.